src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1363 @@
+           with the same options, where some examples are shown.
+
+        EXAMPLES::
+
+            sage: # needs sirocco
+            sage: H.<x, y, z> = ProjectivePlaneCurveArrangements(QQ)
+            sage: A = H(y^2 + x*z, y + x - z, x)
+            sage: A.fundamental_group().sorted_presentation()
+            Finitely presented group < x0, x1 | x1^-1*x0^-1*x1*x0 >
+            sage: A.meridians()
+            {0: [x1], 1: [x0], 2: [x1^-1*x0^-1*x1^-1]}
+            sage: A = H(y^2 + x*z, z, x)
+            sage: A.fundamental_group()
+            Finitely presented group < x0, x1 | (x1*x0)^2*(x1^-1*x0^-1)^2 >
+            sage: A.meridians()
+            {0: [x0, x1*x0*x1^-1], 1: [x0^-1*x1^-1*x0^-1], 2: [x1]}
+            sage: A = H(y^2 + x*z, z*x, y)
+            sage: A.fundamental_group()
+            Finitely presented group < x0, x1, x2 | x2*x0*x1*x0^-1*x2^-1*x1^-1,
+                                                    x1*(x2*x0)^2*x2^-1*x1^-1*x0^-1*x2^-1*x0^-1 >
+            sage: A.meridians()
+            {0: [x0, x2*x0*x2^-1], 1: [x2, x0^-1*x2^-1*x1^-1*x0^-1], 2: [x1]}
+        """
+        if simplified:
+            computed = self._meridians_simpl
+        else:
+            computed = self._meridians_nonsimpl
+        if computed:
+            return computed
+        self._fundamental_group(simplified=simplified)
+        if simplified:
+            return self._meridians_simpl
+        else:
+            return self._meridians_nonsimpl
+
+
+class PlaneCurveArrangements(UniqueRepresentation, Parent):
+    """
+    Plane curve arrangements.
+
+    INPUT:
+
+    - ``base_ring`` -- ring; the base ring
+
+    - ``names`` -- tuple of strings; the variable names
+
+    EXAMPLES::
+
+        sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+        sage: H(x, y^2, x-1, y-1)
+        Arrangement (x, y^2, x - 1, y - 1) in Affine Space
+        of dimension 2 over Rational Field
+    """
+    Element = PlaneCurveArrangementElement
+
+    @staticmethod
+    def __classcall__(cls, base, names=()):
+        """
+        Normalize names
+
+        TESTS::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: K = AffinePlaneCurveArrangements(QQ, names=('x', 'y'))
+            sage: H is K
+            True
+            sage: TestSuite(K).run()
+        """
+        names = normalize_names(len(names), names)
+        return super().__classcall__(cls, base, names)
+
+    def __init__(self, base_ring, names=tuple()):
+        """
+        Initialize ``self``.
+
+        TESTS::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: TestSuite(H).run()
+            sage: H.<x, y, z> = ProjectivePlaneCurveArrangements(QQ)
+            sage: K = ProjectivePlaneCurveArrangements(QQ, names=('x', 'y', 'z'))
+            sage: H is K
+            True
+            sage: type(K)
+            <class 'sage.schemes.curves.plane_curve_arrangement.ProjectivePlaneCurveArrangements_with_category'>
```

Comment:
This is the `type(K)` test I was referring to in my other comment. However, these tests are not part of this `__init__` and are handled elsewhere (and better placed there too). So these should all be removed.
```suggestion
```